generic_scripts/nlk.py

```diff
@@ -0,0 +1,167 @@
+import numpy as _np
+import matplotlib.pyplot as _plt
+from mathphys.constants import vacuum_permeability
+from mathphys.beam_optics import beam_rigidity
+
+
+class Wire:
+    """."""
+    def __init__(self, position, current):
+        self.pos = position  # Column vector
+        self.curr = current
+
+    def Bfield(self, x, y):
+        mu0 = vacuum_permeability
+        r_w = self.pos  # Wire positions
```

Comment:
This could be a source of errors. You assume here that `self.pos` is a column vector, but the setter does not force this shape. I think you should do: `r_w = self._pos[:, None]` here.

Reply:
I think that would be better to follow the idea of forcing the shape in the setter and raise an exception if the array isn't of size 2.

Reply:
I don't know about forcing the shape, but you could force the size of the first dimension to be 2.